### PR TITLE
Tailwind CSS setup

### PR DESCRIPTION
**Pull request by imjuangarcia**
This PR installs and configures [Tailwind CSS](https://tailwindcss.com/) with our Streamlit defaults. The idea is to slowly start refactoring the docs' CSS, moving away from Sass.

Since it'll be a rather complex task and there's no timeline for this project yet, I'm submitting this PR with the configuration, so I can start refactoring components whenever I have some time to spare. This PR shouldn't introduce any changes in styling or functionality.

### CODE REVIEW

**Review by snehankekre — APPROVED** [review]
LGTM, pending @tvst's approval and comments.

**Review by tvst — APPROVED** [review]

**Pull request merged**
